### PR TITLE
[SPARK-18045][SQL][TESTS] Move `HiveDataFrameAnalyticsSuite` to package `sql`

### PR DESCRIPTION
**Pull request by jiangxb1987**
## What changes were proposed in this pull request?

The testsuite `HiveDataFrameAnalyticsSuite` has nothing to do with HIVE, we should move it to package `sql`.
The original test cases in that suite are splited into two existing testsuites: `DataFrameAggregateSuite` tests for the functions and ~~`SQLQuerySuite`~~`SQLQueryTestSuite` tests for the SQL statements.
## How was this patch tested?

~~Modified `SQLQuerySuite` in package `sql`.~~
Add query file for `SQLQueryTestSuite`.

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDataFrameAnalyticsSuite.scala [hunk 1]
```diff
@@ -1,72 +0,0 @@
...
-    try {
-      spark.catalog.dropTempView("mytable")
-    } finally {
-      super.afterAll()
-    }
-  }
-
-  test("rollup") {
-    checkAnswer(
-      testData.rollup($"a" + $"b", $"b").agg(sum($"a" - $"b")),
```
Comment (jiangxb1987) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDataFrameAnalyticsSuite.scala]: Rollup/Cube functions have been tested in `DataFrameAggregateSuite`.

**Comment by jiangxb1987** [comment]
@rxin I've moved the testcases added in this PR to an query file test, do we need to move other test cases for `ROLLUP/CUBE/GROUPING-SETS` too? Currently in `SQLQuerySuite` we have the following:
```
test("rollup")
test("grouping sets when aggregate functions containing groupBy columns") 
test("cube")
test("grouping sets")
test("grouping and grouping_id")
test("grouping and grouping_id in having")
test("grouping and grouping_id in sort")
```

**Comment by rxin** [comment]
It'd be great to move those as well!

**Comment by rxin** [comment]
I'm going to merge this first. Please submit a follow-up pr to move other ones.

**Comment by rxin** [comment]
(Actually I'm on an airplane and the internet is not good enough to merge patches).

**Comment by rxin** [comment]
cc @cloud-fan can you merge this?

**Comment by hvanhovell** [comment]
Merging to master. Thanks! Ping me for the follow-up.

**Pull request closed without merging**